# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1504752871 18000
#      Wed Sep 06 21:54:31 2017 -0500
# Node ID 46dc1ba54f7f494e22be0682c1de1db0f3a863f7
# Parent  387be1a6c3dc1639cef69d00bf28cb1ce6b3c67c
Free variable editor QMainWindow memory upon delete (bug #51954).

* variable-editor.cc (~variable_editor): Delete the m_main window from memory
and, consequently, m_tool_bar and m_tab_widget.

diff --git a/libgui/src/variable-editor.cc b/libgui/src/variable-editor.cc
--- a/libgui/src/variable-editor.cc
+++ b/libgui/src/variable-editor.cc
@@ -124,21 +124,18 @@ variable_editor::variable_editor (QWidge
   setWidget (m_main);
 
   connect (this, SIGNAL (command_requested (const QString&)),
            p, SLOT (execute_command_in_terminal (const QString&)));
 }
 
 variable_editor::~variable_editor (void)
 {
-  // FIXME: This doesn't work yet.
-  //
-  //  delete m_main;
-  //  delete m_tool_bar;
-  //  delete m_tab_widget;
+  // m_tool_bar and m_tab_widget are contained within m_main
+  delete m_main;
 }
 
 void
 variable_editor::edit_variable (const QString& name)
 {
   if (m_stylesheet.isEmpty ())
     {
       QSettings *settings = resource_manager::get_settings ();
